### PR TITLE
Fix the 'ssh: StdinPipe after process started' error

### PR DESCRIPTION
**Pull request by lilirui**
This is a bug can be reproduced in code compiled by Go 1.6.x, the two goroutines execution order can be changed, if the SCP actions start before the StdinPipe, the error 'ssh: StdinPipe after the process started' will be thrown.
The fix is just moving the StdinPipe out of the goroutine, so we can make sure it will be called in a correct order.

@mbhinder @zquestz @variadico @y0ssar1an

### CODE REVIEW

File: ssh/ssh.go [hunk 1]
```diff
@@ -237,17 +237,22 @@ func (client *SSHClient) Download(dst io.WriteCloser, remotePath string) error {
 	wg := sync.WaitGroup{}
 	wg.Add(3)
 
+	ackPipe, err := session.StdinPipe()
+	if err != nil {
+		errorChan <- err
```
Comment (mbhinder) [inline ssh/ssh.go]: return err, also this code could be moved up to line 234 so we don't create the WaitGroup or error channel if something bad happens here.

File: ssh/ssh.go [hunk 2]
```diff
@@ -237,17 +237,22 @@ func (client *SSHClient) Download(dst io.WriteCloser, remotePath string) error {
...
+	}
+
+	defer ackPipe.Close()
+
+	dataPipe, err := session.StdoutPipe()
+	if err != nil {
+		errorChan <- err
```
Comment (mbhinder) [inline ssh/ssh.go]: Same as above

File: ssh/ssh.go (hunk 1 above)
Comment (lilirui) [inline ssh/ssh.go]: Good catch, moved.

File: ssh/ssh.go (hunk 2 above)
Comment (lilirui) [inline ssh/ssh.go]: Done

File: ssh/ssh.go [hunk 3]
```diff
@@ -232,6 +232,18 @@ func (client *SSHClient) Download(dst io.WriteCloser, remotePath string) error {
...
 	defer session.Close()
 
+	ackPipe, err := session.StdinPipe()
+	if err != nil {
+		return err
+	}
+
+	defer ackPipe.Close()
+
+	dataPipe, err := session.StdoutPipe()
```
Comment (mbhinder) [inline ssh/ssh.go]: Any reason we are not closing the dataPipe as well?

**Comment by mbhinder** [comment]
LGTM, we should probably test this manually

File: ssh/ssh.go (hunk 3 above)
Comment (lilirui) [inline ssh/ssh.go]: No such method :)

**Comment by y0ssar1an** [comment]
LGTM

**Pull request merged**